packages/table/src/layers/guides.tsx

```diff
@@ -24,6 +25,17 @@ export interface IGuideLayerProps extends IProps {
 }
 
 export class GuideLayer extends React.Component<IGuideLayerProps, {}> {
+    public shouldComponentUpdate(nextProps: IGuideLayerProps) {
+        if (this.props.className !== nextProps.className) {
+            return true;
+        }
+        // shallow-comparing guide arrays leads to tons of unnecessary re-renders, so we check the
+        // array contents explicitly.
+        const verticalGuidesEqual = Utils.arraysEqual(this.props.verticalGuides, nextProps.verticalGuides);
+        const horizontalGuidesEqual = Utils.arraysEqual(this.props.horizontalGuides, nextProps.horizontalGuides);
+        return !verticalGuidesEqual || !horizontalGuidesEqual;
```

Comment:
may be a minor optimization here to inline both vars in the `||`. then if the first one is true it won't compute the second one (i think that's how js works).

```ts
return !Utils.arraysEqual(this.props.verticalGuides, nextProps.verticalGuides) 
    || !Utils.arraysEqual(this.props.horizontalGuides, nextProps.horizontalGuides);
```